rnote-ui/src/appwindow/appwindowactions.rs

```diff
@@ -537,6 +540,31 @@ impl RnAppWindow {
             }),
         );
 
+        // Remove page from doc in fixed size mode
+        action_remove_page_from_doc.connect_activate(
+            clone!(@weak self as appwindow => move |_action_remove_page_from_doc, _target| {
+                let canvas = appwindow.active_tab().canvas();
+
+                let mut widget_flags = canvas.engine().borrow_mut().record(Instant::now());
+
+                let format_height = canvas.engine().borrow().document.format.height;
+                let format_width = canvas.engine().borrow().document.format.width;
+                let doc_height = canvas.engine().borrow().document.height;
+                let new_doc_height = doc_height - format_height;
+                if doc_height > format_height {
+                    let aabb = Aabb::new(na::point![0.0, new_doc_height], na::point![0.0 + format_width, doc_height]);
+                    let (_, wf) = canvas.engine().borrow_mut().store.split_colliding_strokes(aabb, aabb);
+                    widget_flags.merge(wf);
+                    appwindow.handle_widget_flags(widget_flags, &canvas);
```

Comment:
we should move this to outside of the condition, as the last step of the action

Reply:
Which parts specifically? Only the part that handles stroke splitting and widget flags or all 4 lines of code? Do you mean by 'condition' the if statement?

Reply:
only:

`appwindow.handle_widget_flags(widget_flags, &canvas);`

since that is "produced" by the record call and should always be handled.

> Do you mean by 'condition' the if statement?

yes, to the bottom right before the action closure is finished

Reply:
Done